File: backend/app/schemas/analytics.py
"""数据分析 schema。对应 F4 / T8。"""
from datetime import date

from pydantic import BaseModel


class AnalyticsOverviewQuery(BaseModel):
    start: date | None = None
    end: date | None = None


class AnalyticsOverview(BaseModel):
    recruitment_cycle_days: float | None   # 招聘周期（平均天数）
    conversion_rate: dict                   # 漏斗转化率：各阶段
    channel_effectiveness: dict              # 渠道效果（MVP：按创建来源占位统计）
    total_jobs: int
    total_resumes: int
    total_interviews: int

File: frontend/src/api/analytics.ts
import { api } from "./client";

export interface AnalyticsOverview {
  recruitment_cycle_days: number | null;
  conversion_rate: Record<string, number>;
  channel_effectiveness: Record<string, { uploaded: number; recommended: number; recommended_rate: number }>;
  total_jobs: number;
  total_resumes: number;
  total_interviews: number;
}

export async function getOverview(start?: string, end?: string): Promise<AnalyticsOverview> {
  const resp = await api.get<AnalyticsOverview>("/analytics/overview", { params: { start, end } });
  return resp.data;
}
